flutter-idea/src/io/flutter/deeplinks/DeepLinksViewFactory.java

```diff
@@ -27,18 +28,63 @@
 public class DeepLinksViewFactory implements ToolWindowFactory {
   @NotNull private static String TOOL_WINDOW_ID = "Flutter Deep Links";
 
+  @NotNull
+  private final ViewUtils viewUtils = new ViewUtils();
+
   @Override
   public Object isApplicableAsync(@NotNull Project project, @NotNull Continuation<? super Boolean> $completion) {
-    FlutterSdk sdk = FlutterSdk.getFlutterSdk(project);
-    FlutterSdkVersion sdkVersion = sdk == null ? null : sdk.getVersion();
-    return sdkVersion != null && sdkVersion.canUseDeepLinksTool();
+    // Due to https://github.com/flutter/flutter/issues/142521, this always returns true when the
+    // Flutter IJ plugin is installed.
+
+    // The logic which asserts that the Flutter SDK is up to date enough for this particular feature
+    // is captured in the implementation of createToolWindowContent() below.
+
+    return true;
   }
 
   @Override
   public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
-    FlutterSdk sdk = FlutterSdk.getFlutterSdk(project);
-    FlutterSdkVersion sdkVersion = sdk == null ? null : sdk.getVersion();
+    final FlutterSdk flutterSdk = FlutterSdk.getFlutterSdk(project);
+    final FlutterSdkVersion flutterSdkVersion = flutterSdk == null ? null : flutterSdk.getVersion();
+
+    // There are three potential states for the Flutter SDK:
+    // 1. The Flutter SDK is null (flutterSdk == null) (an SDK path is invalid or not configured)
+    // 2. The Flutter SDK exists, but version file information is unavailable (flutterSdkVersion == null),
+    //      see https://github.com/flutter/flutter/issues/142521.
+    // 3. The Flutter SDK exists, and the version file information is available, but this tool is not
+    //      available on the version of the SDK that the user has.
+    // 4. The Flutter SDK exists with valid version information.
+
+    // First case:
+    if (flutterSdk == null) {
+      viewUtils.presentLabels(toolWindow, List.of("Set the Flutter SDK path in",
+                                                  "Settings > Languages & Frameworks > Flutter,",
+                                                  "and then restart the IDE."));
+      return;
+    }
+
+    // Second case:
+    if (flutterSdk.getVersion().fullVersion().equals(FlutterSdkVersion.UNKNOWN_VERSION)) {
+      viewUtils.presentLabels(toolWindow, List.of("A Flutter SDK was found at the location",
+                                                  "specified in the settings, however the directory",
+                                                  "is in an incomplete state. To fix, shut down the IDE,",
+                                                  "run `flutter doctor` or `flutter --version` in the",
+                                                  "Flutter SDK directory, and then restart the IDE."));
```

Comment:
nit: I don't think this needs to be ran from the flutter sdk directory? `flutter doctor` and `flutter --version` should be able to be run anywhere